crates/iceberg/src/lib.rs

```diff
@@ -73,7 +73,7 @@ mod avro;
 pub mod io;
 pub mod spec;
 
-pub mod metadata_scan;
+pub mod metadata_table;
```

Comment:
Follow-up to this comment: https://github.com/apache/iceberg-rust/pull/822#discussion_r1899617626

Don't need to do in this PR. Up to you.